app/controllers/AuthenticationController.java

```diff
@@ -104,13 +118,28 @@ public Result resetPassword() throws Exception {
      * @param email
      * @return
      */
-    private Study getStudyOrThrowException(Email email) {
-        if (email.getStudyIdentifier() != null) {
-            return studyService.getStudy(email.getStudyIdentifier());
+    @SuppressWarnings("deprecation")
+    private Study getStudyOrThrowException(JsonNode node) {
```

Comment:
This and getStudyIdentifierOrThrowException() have some common code. Would it be better to refactor this to a getStudyIdString() with the fall back logic and have both getStudy() and getStudyId() call it?